### PR TITLE
feat: replace poor pom.xml support with pom dependency tree support

### PR DESCRIPTION
**Pull request by mmcqd**
PR checklist:

- [ ] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [ ] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
ping

File: cli/src/semdep/parse_lockfile.py [hunk 1]
```diff
@@ -661,14 +661,43 @@ def parse_manifest(text: str) -> Set[str]:
...
+        return package_index // 3 + 1
+
+    lines = tree_str.split("\n")
+    deps = [l for l in lines[1:] if l]  # Filter any empty lines
+    for i, dep in enumerate(deps):
+        j = package_index(dep)
+        dep = dep[j:]
+        transitivity = Transitivity(Direct() if depth(j) else Transitive())
+        [_, package, _, version, _] = dep.strip().split(":")
+        print(package, version)
```
Comment (brendongo) [inline cli/src/semdep/parse_lockfile.py]:
```suggestion
```

File: cli/src/semdep/parse_lockfile.py [hunk 2]
```diff
@@ -661,14 +661,43 @@ def parse_manifest(text: str) -> Set[str]:
...
+        i = 0
+        while line[i] in ["+", "-", " ", "\\"]:
+            i += 1
+        return i
+
+    def depth(package_index: int) -> int:
+        return package_index // 3 + 1
+
+    lines = tree_str.split("\n")
+    deps = [l for l in lines[1:] if l]  # Filter any empty lines
```
Comment (brendongo) [inline cli/src/semdep/parse_lockfile.py:614]: Why do you skip first line?

File: cli/src/semdep/parse_lockfile.py [hunk 3]
```diff
@@ -661,14 +661,43 @@ def parse_manifest(text: str) -> Set[str]:
...
+    for i, dep in enumerate(deps):
+        j = package_index(dep)
```
Comment (brendongo) [inline cli/src/semdep/parse_lockfile.py:616]: You can use .lstrip()?

File: cli/src/semdep/parse_lockfile.py (hunk 2 above)
Comment (mmcqd) [reply cli/src/semdep/parse_lockfile.py:614]: It's the name of the project, but there should be a comment saying that

File: cli/src/semdep/parse_lockfile.py (hunk 3 above)
Comment (brendongo) [reply cli/src/semdep/parse_lockfile.py:616]: Ah I see you needed the index later

Comment (mmcqd) [reply cli/src/semdep/parse_lockfile.py:616]: It would have to strip multiple different characters

File: cli/src/semdep/parse_lockfile.py [hunk 4]
```diff
@@ -661,14 +661,43 @@ def parse_manifest(text: str) -> Set[str]:
...
 
+def parse_pom_tree(tree_str: str, _: Optional[str]) -> Iterator[FoundDependency]:
+    def package_index(line: str) -> int:
+        i = 0
+        while line[i] in ["+", "-", " ", "\\"]:
+            i += 1
+        return i
+
+    def depth(package_index: int) -> int:
+        return package_index // 3 + 1
```
Comment (brendongo) [inline cli/src/semdep/parse_lockfile.py:610]: Are you sure this is consistent?

File: cli/src/semdep/parse_lockfile.py (hunk 3 above)
Comment (brendongo) [reply cli/src/semdep/parse_lockfile.py:616]: lstrip does "abc123".lstrip("cab") returns "123"

**Review by brendongo — APPROVED** [review]

File: cli/src/semdep/parse_lockfile.py (hunk 4 above)
Comment (mmcqd) [reply cli/src/semdep/parse_lockfile.py:610]: I should make the example file bigger but yes I am

**Pull request merged**